driver/utils/type_info.h

```diff
@@ -0,0 +1,2114 @@
+
+                        if (dest.value != 0) {
+                            if ((dest_value_max / dec_mult) < dest.value)
+                                throw std::runtime_error("Cannot interpret '" + src + "' as Decimal/Numeric: value is too big for internal representation");
+                            dest.value *= dec_mult;
+                        }
+
+                        if (next_dec_dig != 0) {
+                            if ((dest_value_max - next_dec_dig) < dest.value)
+                                throw std::runtime_error("Cannot interpret '" + src + "' as Decimal/Numeric: value is too big for internal representation");
+                            dest.value += next_dec_dig;
+                        }
+
+                        if (dot_met)
+                            ++right_n;
+                        else
+                            ++left_n;
+
+                        dig_met = true;
+                        break;
+                    }
+
+                    default:
+                        throw std::runtime_error("Cannot interpret '" + src + "' as Decimal/Numeric");
+                }
+            }
+
+            if (dest.value == 0)
+                dest.sign = 1;
+
+            dest.precision = left_n + right_n;
+            dest.scale = right_n;
+        }
+    };
+
+    template <>
+    struct from_value<std::string>::to_value<DataSourceType<DataSourceTypeId::Decimal32>> {
+        using DestinationType = DataSourceType<DataSourceTypeId::Decimal32>;
+
+        static inline void convert(const SourceType & src, DestinationType & dest) {
+            return to_value<DataSourceType<DataSourceTypeId::Decimal>>::convert(src, dest);
+        }
+    };
+
+    template <>
+    struct from_value<std::string>::to_value<DataSourceType<DataSourceTypeId::Decimal64>> {
+        using DestinationType = DataSourceType<DataSourceTypeId::Decimal64>;
+
+        static inline void convert(const SourceType & src, DestinationType & dest) {
+            return to_value<DataSourceType<DataSourceTypeId::Decimal>>::convert(src, dest);
+        }
+    };
+
+    template <>
+    struct from_value<std::string>::to_value<DataSourceType<DataSourceTypeId::Decimal128>> {
+        using DestinationType = DataSourceType<DataSourceTypeId::Decimal128>;
+
+        static inline void convert(const SourceType & src, DestinationType & dest) {
+            return to_value<DataSourceType<DataSourceTypeId::Decimal>>::convert(src, dest);
+        }
+    };
+
+    template <>
+    struct from_value<std::int64_t> {
+        using SourceType = std::int64_t;
+
+        template <typename DestinationType>
+        struct to_value {
+            static inline void convert(const SourceType & src, DestinationType & dest) {
+                if constexpr (std::is_convertible_v<SourceType, DestinationType>) {
+                    dest = src;
+                }
+                else {
+                    convert_via_proxy<std::string>(src, dest);
```

Comment:
This, and every other `convert_via_proxy<std::string>` looks spooky and very sub-optimal, luckily these branches are never hit: https://github.com/traceon/clickhouse-odbc/pull/37/commits/cf58fff2f08222b8360b3a1ecf761f00a6bec67d

And can be safely removed.